src/effects/deeplearning/ModelManagerPanel.cpp

```diff
@@ -0,0 +1,809 @@
+               this->mInstallProgressGauge->SetRange(expected);
+               this->mInstallProgressGauge->SetValue(current);
+            }
+            else
+               this->mInstallProgressGauge->Pulse();
+         });
+   });
+
+   CompletionHandler onInstallDone([this, &manager](int httpCode, std::string responseBody)
+   {  
+      this->CallAfter(
+         [this, httpCode, &manager]()
+         {
+            if (httpCode == 200 || httpCode == 302)
+            {
+               // check if install succeeded
+               if (manager.IsInstalled(this->mCard))
+                  this->SetInstallStatus(InstallStatus::installed);
+               else
+               {
+                  this->mEffect->Effect::MessageBox(
+                     XO("An error ocurred while installing the model with Repo ID %s. ")
+                        .Format(this->mCard->GetRepoID())
+                  );
+                  this->SetInstallStatus(InstallStatus::uninstalled);
+               }
+            }
+            else
+            {
+               this->SetInstallStatus(InstallStatus::uninstalled);
+               this->mEffect->Effect::MessageBox(
+                  XO("An error ocurred while downloading the model with Repo ID %s. \n"
+                     "HTTP Code: %d").Format(this->mCard->GetRepoID(), httpCode)
+               );
+            }
+         }
+      ); 
+   });
+
+   if (!manager.IsInstalled(mCard))
+   {
+      this->SetInstallStatus(InstallStatus::installing);
+
+      // TODO: since this is done in another thread, how do I catch an error, like
+      // losing the connection in the middle of a download? 
+      manager.Install(mCard, std::move(onProgress), std::move(onInstallDone));
+   }
+}
+
+void ModelCardPanel::OnSelect(wxCommandEvent &event)
+{  
+   mEffect->SetModel(mCard);
+}
+
+void ModelCardPanel::OnEnable(wxCommandEvent &event)
+{
+   mEffect->SetModel(mCard);
+}
+
+void ModelCardPanel::OnMoreInfo(wxCommandEvent &event)
+{
+   DeepModelManager &manager = DeepModelManager::Get();
+   wxString url = wxString(manager.GetMoreInfoURL(mCard));
+   wxLaunchDefaultBrowser(url);
+}
+
+void ModelCardPanel::SetModelStatus(ModelStatus status)
+{
+   if (status == ModelStatus::enabled)
+   {
+      SetBackgroundColour(
+         theTheme.Colour(clrMediumSelected)
+      );
+   }
+   else
+   {
+      SetBackgroundColour(
+         theTheme.Colour(clrMedium)
+      );
+   }
+
+   // Fit();
+   Refresh();
+}
+
+void ModelCardPanel::OnClick(wxMouseEvent &event)
+{
+   mEffect->SetModel(mCard);
+}
+
+// SimpleModelCardPanel
+
+SimpleModelCardPanel::SimpleModelCardPanel(wxWindow *parent, wxWindowID id,
+                           ModelCardHolder card, EffectDeepLearning *effect, ModelManagerPanel *managerPanel)
+      : ModelCardPanel(parent, id, card, effect, managerPanel, wxSize(MODELCARDPANEL_WIDTH, MODELCARDPANEL_HEIGHT))
+{
+   Populate();
+}
+
+
+void SimpleModelCardPanel::PopulateOrExchange(ShuttleGui &S)
```

Comment:
Different set of h/cpp?

Reply:
https://github.com/audacity/audacity/pull/1384/commits/8d5285a0c5c9ff3d18c46ad23c78a0a50859370c